weave/scorers/guardrails/restricted_terms_guardrail.py

```diff
@@ -0,0 +1,159 @@
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+from pydantic import BaseModel, Field
+
+import weave
+from weave.scorers import Scorer
+from weave.scorers.guardrails.prompts import (
+    RESTRICTED_TERMS_GUARDRAIL_SYSTEM_PROMPT,
+    RESTRICTED_TERMS_GUARDRAIL_USER_PROMPT,
+)
+from weave.scorers.llm_utils import OPENAI_DEFAULT_MODEL, create
+
+if TYPE_CHECKING:
+    from instructor import Instructor
+
+
+class TermMatch(BaseModel):
+    """Represents a matched term and its variations"""
+
+    original_term: str
+    matched_text: str
+    match_type: str = Field(
+        description="Type of match: EXACT, MISSPELLING, ABBREVIATION, or VARIANT"
+    )
+    explanation: str = Field(
+        description="Explanation of why this is considered a match"
+    )
+
+
+class RestrictedTermsAnalysis(BaseModel):
+    """Analysis result for restricted terms detection"""
+
+    contains_restricted_terms: bool = Field(
+        description="Whether any restricted terms were detected"
+    )
+    detected_matches: list[TermMatch] = Field(
+        default_factory=list,
+        description="List of detected term matches with their variations",
+    )
+    explanation: str = Field(description="Detailed explanation of the analysis")
+    anonymized_text: Optional[str] = Field(
+        default=None,
+        description="Text with restricted terms replaced with category tags",
+    )
+
+    @property
+    def safe(self) -> bool:
+        return not self.contains_restricted_terms
+
+
+class RestrictedTermsRecognitionResponse(BaseModel):
+    """Response from the `RestrictedTermsLLMGuardrail`"""
+
+    safe: bool
+    detected_entities: list[TermMatch]
+    reasoning: str
+    anonymized_text: Optional[str] = None
+
+
+class RestrictedTermsLLMGuardrail(Scorer):
+    """A guardrail to detect and analyze restricted terms and their variations in a given
+    text prompt using an LLM.
+
+    Attributes:
+        system_prompt (str): The prompt describing the task of detecting restricted terms.
+        user_prompt (str): The prompt template to pass the input and output data. The template must
+            contain placeholders for both `{text}` and `{custom_terms}`.
+        model_id (str): The LLM model name, depends on the LLM's providers to be used `client` being used.
+        temperature (float): LLM temperature setting.
+        max_tokens (int): Maximum number of tokens in the LLM's response.
+        should_anonymize (bool): Whether to anonymize the text.
+        custom_terms (list[str]): The list of custom terms to detect.
+    """
+
+    system_prompt: str = RESTRICTED_TERMS_GUARDRAIL_SYSTEM_PROMPT
+    user_prompt: str = RESTRICTED_TERMS_GUARDRAIL_USER_PROMPT
+    model_id: str = OPENAI_DEFAULT_MODEL
+    temperature: float = 0.1
+    max_tokens: int = 4096
+    should_anonymize: bool = True
+    custom_terms: list[str] = [
```

Comment:
I'd remove the defaults we have here

Reply:
I think it's better to retain default values for `temperature` and `max_tokens` to ensure coherence with the other scorers. WDYT @ash0ts?

Reply:
Oh I meant the defaults for the custom_terms. The rest are good to me